feat: Added Search Functionality & Improved Dark-Light Mode Toggle

RohanKrMahato opened the pull request:
# Add Search Functionality & Improve Dark-Light Mode Toggle  

## Summary  
This PR introduces a **Search functionality** to the documentation and enhances the **Dark-Light mode toggle** for a smoother user experience.  

## Changes & Fixes  
- **Added** a **Search feature** to improve content discoverability.  
- **Fixed** Dark-Light mode toggle issues for better theme switching.  

## Screenshots  

### Before Adding Search Functionality  
![Before Search](https://github.com/user-attachments/assets/82d598e5-fa8c-4a47-8dda-e954870ad466)  

### After Adding Search Functionality  
![After Search 1](https://github.com/user-attachments/assets/bb24212f-9d95-41d3-ba80-079b41a48afc)  
-
![After Search 2](https://github.com/user-attachments/assets/f2966047-1be9-42f1-bc2a-776d1709eddf)  

---  

### Before Improving Dark-Light Mode Toggle  
![Before Toggle Fix](https://github.com/user-attachments/assets/3547da68-9b5d-47bc-8d36-22426244a628)  

### After Improving Dark-Light Mode Toggle  
![After Toggle Fix](https://github.com/user-attachments/assets/bdff2ec1-c8f6-4363-86ad-bdc61957a712)  

## How to Test  
1. **Search Feature**  
   - Type keywords in the search bar and verify results.  

2. **Dark-Light Mode Toggle**  
   - Switch between Dark and Light modes to ensure smooth transitions.

ltorres6 reviewed (dismissed):
The mkdocs search is terrible in my opinion, but I suppose it's better than no search. Thanks!

ltorres6 commented:
@RohanKrMahato Can you update the poetry toml file as well to align the lock file and fix the ci/cd check?

RohanKrMahato commented:
> @RohanKrMahato Can you update the poetry toml file as well to align the lock file and fix the ci/cd check?

Sure sir! I will work on it.

MohamedNasser8 commented:
@ltorres6 Should we change the color to match the OSIPI pallet here?
Or do it in a separate PR.

![image](https://github.com/user-attachments/assets/d206a0f3-f0b5-4dc8-8e93-dbcc5fe8879c)

RohanKrMahato commented:
@ltorres6  Sir, I have fixed the pre-commit issue that was occurring due to formatting in the mkdocs.yml file. I have also updated the lock file using `poetry lock`.

Please let me know if any further changes are needed. Thank you!

RohanKrMahato commented:
@ltorres6 It looks like the poetry lock check is failing again. This might be due to the CI using a specific version of Poetry. Should I update it to poetry@latest?

MohamedNasser8 commented:
Hey @RohanKrMahato 

- Run:
`
poetry lock --no-update
`

- Then run:
`
poetry install
`
then push your changes it should work.

RohanKrMahato commented:
@MohamedNasser8 thanks bro, trying it now. I will let you know if it worked.

RohanKrMahato commented:
![image](https://github.com/user-attachments/assets/bd851a17-be38-4686-850e-1cd2860345a3)

also, if I simply run `poetry lock` then the lock file get updated to the latest version of poetry ( 2.1.1 )

ltorres6 reviewed (approved):
LGTM

Thanks @RohanKrMahato and @MohamedNasser8. Downgrading locally is fine for this and I will merge.

@MohamedNasser8 We should probably just update poetry in the CI/CD.

MohamedNasser8 commented:
@ltorres6 I think it's not optimal to use the latest version.
I think I'll extract the version from the lock file and install poetry in that version so no conflict occurs.
